relay-server/src/services/project.rs

```diff
@@ -585,6 +585,19 @@ impl Project {
             return;
         };
 
+        Self::apply_metrics_deny_list(&state.config.deny_metrics, metrics);
+        Self::filter_disabled_namespace(state, metrics);
+    }
+
+    fn apply_metrics_deny_list(deny_list: &ErrorBoundary<Metrics>, buckets: &mut Vec<Bucket>) {
```

Comment:
_nit_: Would prefer to inline this function. It's not an associated function as it doesn't rely on the project, but you're passing fields of the project in. Since the function is rather small, it would lend itself for inlining:

```rust
if let ErrorBoundary::Ok(ref metrics) = state.config.deny_metrics.metrics {
  // TODO: Move filter_metrics into this file
  metrics.filter_metrics(buckets); 
}
```

Reply:
rather than inling, I moved more responsibility to it, since I removed hte associated method on `Metrics` to avoid the `Bucket` dependency in the config crate. 
 
I assume thats's what you meant earlier, unless I should have multiple impl blocks to get around the dependency problem?